ebean-datasource/src/main/java/io/ebean/datasource/pool/ConnectionPool.java

```diff
@@ -725,6 +733,44 @@ private void stopHeartBeatIfRunning() {
     }
   }
 
+  void execute(Runnable runable) {
+    executorLock.lock();
+    try {
+      if (executor != null) {
+        executor.submit(runable);
+        return;
+      }
+    } finally {
+      executorLock.unlock();
+    }
+    // it is possible, that we receive runnables after shutdown.
+    // in this case, we will execute them immediately (outside lock)
+    runable.run();
+  }
+
+  private void startExecutor() {
+    executorLock.lock();
+    try {
+      if (executor == null) {
+        executor = Executors.newSingleThreadExecutor();
```

Comment:
Here I configure the executor. Currently I think a singleThreadExecutor is enough.
Theoretically also a multi-thread executor can be used here to support closing multiple connections in parallel.
CHECKME: Is this OK? I could also introduce a config-value e.g. `closingThreads=1` - a value of `0` would restore the old behaviour (closing connections synchroniously). Should I do that?

Reply:
> Currently I think a singleThreadExecutor is enough.

Happy enough with this. For Java 21+ this could be a virtualThreadPerTaskExecutor ... so we could do that with a Multi-Version jar (I could add that part)

> value of 0 would restore the old behaviour (closing connections synchronously)

Hmm. My gut says lets not do that at least initially - keep it simple / don't see too many benefits in closing connections synchronously at this stage.

Reply:
Looking at going with: https://github.com/ebean-orm/ebean-datasource/pull/134/commits/07f9807c812d7febadc3d4a1d99d832130f3e3e9